pandas/core/frame.py

```diff
@@ -337,6 +338,10 @@
       join; sort keys lexicographically.
     * inner: use intersection of keys from both frames, similar to a SQL inner
       join; preserve the order of the left keys.
+    * left_semi: Filter for rows in the left that have a match on the right;
+      preserve the order of the left keys.
```

Comment:
These sound really similar. I think the difference is that if a key appears more than once on the left, then only one match is returned.  Secondly, no columns from the right are returned.  Did I get that right?  If so, I think you may want to say more here.